environment/src/main/java/com/sequenceiq/environment/environment/validation/validators/EncryptionKeyArnValidator.java

```diff
@@ -25,4 +52,34 @@ public ValidationResult validateEncryptionKeyArn(String encryptionKeyArn) {
         }
         return validationResultBuilder.build();
     }
+
+    public ValidationResult validate(EnvironmentValidationDto environmentValidationDto, ParametersDto parametersDto) {
+        EnvironmentDto environmentDto = environmentValidationDto.getEnvironmentDto();
+        ValidationResult.ValidationResultBuilder validationResultBuilder = ValidationResult.builder();
+        String encryptionKeyArn = parametersDto.getAwsParametersDto().getAwsDiskEncryptionParametersDto().getEncryptionKeyArn();
+        CloudCredential cloudCredential = credentialToCloudCredentialConverter.convert(environmentDto.getCredential());
+        Region region = region(environmentDto.getLocation().getName());
+
+        CloudEncryptionKeys encryptionKeys = new CloudEncryptionKeys(new HashSet<>());
+
+        try {
+            CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(
+                    Platform.platform(environmentDto.getCloudPlatform()), null);
+            encryptionKeys = cloudPlatformConnectors.get(cloudPlatformVariant)
+                    .platformResources().encryptionKeys(cloudCredential, region, Collections.emptyMap());
+        } catch (RuntimeException e) {
+            LOGGER.error("Failed to fetch AWS Cloud keys", e.getMessage());
+
```

Comment:
We need to give different error message to the user right? This is not a user input problem yet, just that we are not able to validate it, may be due to many reasons like API throttling, network error etc.

Reply:
`"An unexpected error occurred while trying to fetch the AWS cloud keys", e.getMessage()` Could you please confirm if this error message makes more sense to the user than the previous one as the e.getmessage would have the error details for user to refer to ?

Reply:
@satyabollineni We are swallowing the exception here. This means the error message we are sending to the user will have "The provided encryption key does not exist in the given region's encryption key list for this credential.". from the code below. So we are going to give an error message that says two things, what the problem was and what the user could do next. For example, including a phrase like please retry the operation will help the user for this case. Also, according to https://jira.cloudera.com/browse/CB-12758?focusedCommentId=4749072&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4749072 these types of validation failures (AWS throttling) should result in cluster creation to proceed and the user gets an indication in events that the validation was not fully complete. CC: @lnardai

Reply:
@cegganesh84 Thanks for elaborating on your comment above. I agree with you on the retry part and I would introduce the retry logic If we face any failures from AWS end like AWS throttling etc. But I have a slightly different view on the second part i.e **"these types of validation failures (AWS throttling) should result in cluster creation to proceed and the user gets an indication in events that the validation was not fully complete."**. If a user passes the encryption keys, User is expecting to encrypt the data but for any reason If the validation fails and we proceed to cluster creation without encryption then that would be contrast to the user expectation of encrypting the data on cluster.

Lets say we proceed without validating the key (if validation failed due to AWS throttling etc) and try to encrypt the data then at the later point If the key is invalid we fail to encrypt which is not expected for the end user. (of course things will go good If luckily the key is valid). In my opinion failing would be more appropriate in these cases rather than proceeding. But I would be happy to know If I am missing any point here. cc @lnardai

Reply:
Yes, we should always fail if the encryption parameter is given but we could not encrypt, there is no different viewpoint there. AWS will give an error message if you pass an invalid encryption arn, and we will rely on it. The issue is only during the validation phase (which is to give an upfront error message) about what to do with transient issues because of which we do not know if the key is valid or not.